proposals/Exceptions.md

```diff
@@ -262,130 +249,475 @@ Like the `block`, `loop`, and `if` instructions, the `try` instruction is a
-exceptions, assigning monotonically-increasing indices based on the order
-defined in the import and exception sections. Thus, the index space starts at
-zero with imported exceptions followed by internally-defined exceptions in
-the [exception section](#exception-section).
+The event index space defines the (module) static version of runtine event tags.
+For event indices that are not imported/exported, the corresponding event tag is
+guaranteed to be unique over all loaded modules. Events that are imported or
+exported alias the respective events defined elsewhere, and use the same tag.
 
 ## Changes to the binary model
 
-This section describes changes in
-the
-[binary encoding design document](https://github.com/WebAssembly/design/blob/master/BinaryEncoding.md).
+This section describes changes in the [binary encoding design
+document](https://github.com/WebAssembly/design/blob/master/BinaryEncoding.md).
 
-### except_type
+#### Other Types
 
-An exception is described by its exception type signature, which corresponds to
-the data fields of the exception.
+##### event_type
+
+The set of event attributes are:
+
+| Name      | Value |
+|-----------|-------|
+| Exception | 0     |
+
+Each event type has the fields:
 
 | Field | Type | Description |
 |-------|------|-------------|
-| `count` | `varuint32` | The number of types in the signature |
-| `type` | `value_type*` | The type of each element in the signature |
-
-### External kind
+| `attribute` | `varuint32` | The attribute of the event. |
+| `type` | `varuint32` | The type index for its corresponding type signature |
 
-`external_kind`
+##### external_kind
 
 A single-byte unsigned integer indicating the kind of definition being imported
 or defined:
 
-* `0` indicating a `Function` [import](Modules.md#imports) or [definition](Modules.md#function-and-code-sections)
-* `1` indicating a `Table` [import](Modules.md#imports) or [definition](Modules.md#table-section)
-* `2` indicating a `Memory` [import](Modules.md#imports) or [definition](Modules.md#linear-memory-section)
-* `3` indicating a `Global` [import](Modules.md#imports) or [definition](Modules.md#global-section)
-* `4` indicating an `Exception` [import](#import-section) or [definition](#exception-sectio)
+* `0` indicating a `Function` [import](Modules.md#imports) or
+[definition](Modules.md#function-and-code-sections)
+* `1` indicating a `Table` [import](Modules.md#imports) or
+[definition](Modules.md#table-section)
+* `2` indicating a `Memory` [import](Modules.md#imports) or
+[definition](Modules.md#linear-memory-section)
+* `3` indicating a `Global` [import](Modules.md#imports) or
+[definition](Modules.md#global-section)
+* `4` indicating an `Event` [import](#import-section) or
+[definition](#event-section)
+
+### Module structure
+
+#### High-level structure
+
+A new `event` section is introduced and is named `event`. If included, it must
+appear after immediately after the global section.
```

Comment:
nit: One redundant "after" (the first one).

Reply:
Done